### PR TITLE
5909 pre-fill Github issue reports with crash details

### LINKED ISSUE #5909: Bug-reporting UX
It would be nice if 
a) the `Submit Bug To GitHub` button were to **copy the reported error to the clipboard** and
b) **open this URL**: https://github.com/DynamoDS/Dynamo/issues/new
capture
(I know I can manually copy the error msg but I keep forgetting and hitting that button before doing so...)
Dynamo 0.9 stable

### PR DESCRIPTION
**Pull request by radumg**
### Purpose

Fixes #5909 by pre-filling the new Github issue page with 
- an appropriate title that includes Dynamo version number : `Crash report from Dynamo 2.0.1`
- issue body filled with the crash details text, displayed in the `CrashPrompt` window bottom text box

Currently pressing `Submit to Github` takes you here : 

image

With the new behaviour, pressing `Submit to Github` takes you here :
 
image

In the above image, the `This is the error text` body would be replaced by the actual details of the crash stack trace, as per original screenshot in #5909 . 

I can't seem to be able to make Dynamo crash on purpose right now (funny how that works...), so dunno how to replicate.

#### Other notes
- sends users to `https://github.com/DynamoDS/Dynamo/issues/new` by default when no crash text is supplied (friendlier than sending to the super-busy all issues page)
- as per current implementation, you still need to be logged in to Github to be able to submit the issue

#### Further improvements
We could add the rest of the current `issue template` text to the body and wrap the `crash dump` inside a nice code block, but I'm afraid that'd make the query string way too long. Open to ideas/suggestions ?

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@Racel 

### FYIs

@andydandy74 , @johnpierson

### CODE REVIEW

**Comment by mjkkirschner** [comment]
This is great, we were just talking about this -additionally would be great to add:
* dynamo version number
* dynamoRevit version number
* packages
* dyn/dyf dependencies if the user wants them to be uploaded...

just things we have discussed, not that all that needs to go in this PR.

🎉

**Comment by radumg** [comment]
Thanks @mjkkirschner !

Dynamo version number is actually included in the issue title.

The rest could also be added to body of issue, my only concern is how feasible it is to use the simple url query params mechanism i'm using here to add that much info - haven't stress-tested it yet 💪

**Comment by andydandy74** [comment]
Looks like 2kB of data is safe:
https://stackoverflow.com/questions/2659952/maximum-length-of-http-get-request

**Comment by alfarok** [comment]
Hey @radumg I just made sure this is in the backlog to get reviewed next sprint.  It would be a nice improvement!

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs [hunk 1]
```diff
@@ -64,4 +64,4 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyFileVersion("2.0.2.6067")]
+[assembly: AssemblyFileVersion("2.0.2.6426")]
```
Comment (alfarok) [inline src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: These changes don't need to be committed and can be excluded

File: src/DynamoCoreWpf/Properties/Resources.Designer.cs [hunk 2]
```diff
@@ -19,7 +19,7 @@ namespace Dynamo.Wpf.Properties {
     // class via a tool like ResGen or Visual Studio.
     // To add or remove a member, edit your .ResX file then rerun ResGen
     // with the /str option, or rebuild your VS project.
-    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "4.0.0.0")]
+    [global::System.CodeDom.Compiler.GeneratedCodeAttribute("System.Resources.Tools.StronglyTypedResourceBuilder", "15.0.0.0")]
```
Comment (alfarok) [inline src/DynamoCoreWpf/Properties/Resources.Designer.cs]: Is this update required?  Are there any other repercussions to updating?

File: src/DynamoCoreWpf/Properties/Resources.resx [hunk 3]
```diff
@@ -1961,10 +1961,10 @@ Do you want to install the latest Dynamo update?</value>
     <value>Package Path Added</value>
   </data>
   <data name="PackagePathAutoAddNotificationShortDescription" xml:space="preserve">
-    <value>A library (*.dll, *.ds) was recently imported into Dynamo. Its path was automatically added to "Settings > Manage Node and Package Paths..."</value>
+    <value>A library (*.dll, *.ds) was recently imported into Dynamo. Its path was automatically added to "Settings &gt; Manage Node and Package Paths..."</value>
```
Comment (alfarok) [inline src/DynamoCoreWpf/Properties/Resources.resx]: Was this change required?

**Comment by alfarok** [comment]
Took a quick peak, I just added a couple comments/questions but looks good.  We will get it pulled next sprint for further testing.

File: src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs (hunk 1 above)
Comment (radumg) [reply src/AssemblySharedInfoGenerator/AssemblySharedInfo.cs]: i'm not doing anything on purpose here (include or exclude) - do you mean I should manually exclude (not commit) them ? Feels like there should be something in `.gitignore` for it ?

File: src/DynamoCoreWpf/Properties/Resources.Designer.cs (hunk 2 above)
Comment (radumg) [reply src/DynamoCoreWpf/Properties/Resources.Designer.cs]: This is due to my VisualStudio 2017, i'll revert to be on the safe side.

File: src/DynamoCoreWpf/Properties/Resources.resx (hunk 3 above)
Comment (radumg) [reply src/DynamoCoreWpf/Properties/Resources.resx]: Visual Studio 2017 did this one, i can revert it.

**Comment by radumg** [comment]
Thanks @alfarok for the review, i'll fix up those issues now 👍

**Comment by radumg** [comment]
should be ready for review again @alfarok !

**Comment by alfarok** [comment]
Thanks for the updates @radumg, looks good.  We will run the changes on the self-serve CI and get some manual testing in the upcoming sprint.  It may also be worth adding a unit test calling `ReportABug` (without actually forcing a crash) just to verify everything is operating as expected.

**Comment by radumg** [comment]
Might get [tricky/messy ](https://stackoverflow.com/a/10588918) to detect a broweser window opened correctly, but i could decompose the window launching & figuring out what to launch part, so we can unit test - would that be better ?

**Comment by alfarok** [comment]
Hey @radumg, I don't think it will be super easy test the issue text body but what about verifying the window is open and has the correct window title.  I would try to call `ReportABug()`, verify the process is running `Process.GetProcesses()`, and use `Process.MainWindowTitle` to verify the expected window exists in the running processes.  Let me know if that makes sense or if you run into any issues.

**Comment by radumg** [comment]
Yep, makes sense and i was thinking similar, so good to know that's all we'd need to check 👍 
Now, because this launches user default browser, we have no way of knowing if it'll launch `Chrome`, `IE`, `Firefox` or `Edge` (or `Opera` for hipster users 😛). I'll look into it, though.

**Comment by alfarok** [comment]
I think no matter which browser gets launched the main window title should be consistent, such as `New Issue - DynamoDS/Dynamo` but I am not positive on that with tabs etc.  Thanks for looking into this.  It would be really great if we could make the self-serve CI publicly accessible in the future to make testing easier for everyone.  Maybe Microsoft will give us some additional new options!

**Comment by alfarok** [comment]
Hey @radumg I just realized this PR is to `RC2.0.2_master` but it should be to `master` (2.1.0), no new features are currently going into 2.0.2.  I just kicked off your branch on the self-serve CI to make sure all the existing tests are passing.  I will also pull it down locally over the next couple days to do a little manual testing.

**Comment by radumg** [comment]
thanks @alfarok , will rebase on `master` then.

**Comment by radumg** [comment]
hmmm, changing base to `master` does not look quite ok, not sure which branch this should be based on then ?
image


I just picked the highest 2.0+ branch i found at the time 😁

**Comment by alfarok** [comment]
@radumg That's because you are diffing your fork of `RC2.0.2_master` against `Master`.  These 2 branches have quite a few difference so you are seeing all of them when comparing.  There are a couple options but the easiest is probably to cherry-pick your commits to master on your fork and open a new PR.  (Let me know if you have any questions on this)

I also have some good news and bad news.  The good news is all the tests are passing with your changes.  The bad news is that pressing the `Submit to Github` button seems to crash the crash?  After clicking, everything closes including Dynamo with no further activity.  Looks like an exception is being thrown:

image

Previously the Dynamo and crash windows would remain open while the browser was opened allowing the user to save upon pressing `Continue`.  This makes that test even more important because we should be getting a failure if the behavior has regressed.  Since the self-serve is passing it means we have no code coverage on this functionality ☹.

The crash I was using to test was reported by @andydandy74 recently.  Load the Core_Math sample, select all the nodes, use node to code.  I also tested on master and the previous implementation launches the browser as expected and Dynamo remains open.

**Comment by alfarok** [comment]
It seems the issue is that `Resources.CrashPromptGithubNewIssueTitle` is returning null

**Comment by radumg** [comment]
Great, thanks so much @alfarok ! I can test this now knowing a reproducible crash.
I'll cherry-pick to new PR if I can't rebase more cleanly.

**Comment by radumg** [comment]
Hey Keith @alfarok , haven’t managed to look at it post-AU (backlog..😖) but will set some time aside on Friday and let you know how I get on 🙂

**Comment by radumg** [comment]
@alfarok , made a new PR 
- based on latest `master` branch
- adds unit test
- fixes `null` check
